packages/flutter_tools/lib/src/device.dart

```diff
@@ -28,6 +28,23 @@ import 'windows/windows_device.dart';
 
 DeviceManager get deviceManager => context.get<DeviceManager>();
 
+/// A broad category the describes the kind of workflow the device supports.
+class Category {
+  const Category._(this.value);
+
+  static const Category web = Category._('web');
+  static const Category fuchsia = Category._('fuchsia');
+  static const Category desktop = Category._('desktop');
+  static const Category mobile = Category._('mobile');
+  static const Category simulator = Category._('simulator');
```

Comment:
Should we call this "emulator" or "simulator"? We stick the iOS simulators in the "flutter emulators" command and generally refer to them as emulators (except for their names). I don't know which type is more common though!